### DISCUSSION TITLE
Please check your URI response entry and try again.

### DISCUSSION THREAD

**Discussion by darrepac** [abraunegg/onedrive#1426] [Q&A] [answered]
Hi

I copy the link, I log into microsoft, I got a blanck page (not error message), I copy/paste the full URL and then I got:
`Application has not been successfully authorised. Please check your URI response entry and try again.`
How can I debug or do differently?

Note: my server is headless, so I am using an other computer in my LAN to do the process. I have seen that it should not be a problem

thanks

**Comment by abraunegg** [comment]
@darrepac 
> I copy the link, I log into microsoft, I got a blanck page (not error message), I copy/paste the full URL and then I got:
`Application has not been successfully authorised. Please check your URI response entry and try again.`
How can I debug or do differently?

The error message that is being displayed is indicative of:
* You are not copying / pasting the correct response URL
* You are not copying / pasting the *entire* URL - the length of the URL will depend on your account type. When providing the URL you have to provide it exactly as it is given
* The response URL contains an error - paste the URL into a text editor and confirm no error
* You are potentially performing the action in in one Geographic region whilst trying to authenticate the client in another - the OneDrive authentication process has had issues with this in the past (cross regional boundaries / timezone issues)
* The time synchronisation between your devices is incorrect / out by > 5 minutes - ensure all your devices are utilising NTP and synced to applicable time sources
* The timezone configuration of your devices is incorrect for your geographic region
* Your device that is attempting to authorise is unable to access the OneDrive API (firewall / DNS errors)
* Your device that is attempting to authorise is unable to query DNS within the required timeframe (slow DNS)

Suggest you review your actions, your process & your entire setup and try again.

**Reply by darrepac** [reply]
Thanks for the list
> 
> * You are not copying / pasting the correct response URL

=> well I don't see how I can be wrong, the process seems simple

> * You are not copying / pasting the _entire_ URL - the length of the URL will depend on your account type. When providing the URL you have to provide it exactly as it is given

=> checked 

> * The response URL contains an error - paste the URL into a text editor and confirm no error

=> done, no error


> * You are potentially performing the action in in one Geographic region whilst trying to authenticate the client in another - the OneDrive authentication process has had issues with this in the past (cross regional boundaries / timezone issues)

=> I am in France with french IP with french onedrive account (=from french university)


> * The time synchronisation between your devices is incorrect / out by > 5 minutes - ensure all your devices are utilising NTP and synced to applicable time sources

=> checked

> * The timezone configuration of your devices is incorrect for your geographic region

=> checked (Europe/Paris, correct)

> * Your device that is attempting to authorise is unable to access the OneDrive API (firewall / DNS errors)

=> checked (I got no error but to be sure I deactivate my local DNS, to use my ISP one)

> * Your device that is attempting to authorise is unable to query DNS within the required timeframe (slow DNS)

=> I do not see any reason for that

> Suggest you review your actions, your process & your entire setup and try again.

=> well, I am at the end of the list and I review everything and I see nothing wrong. My sole doubt is the type of onedrive account that I have (it is a university account) but I succesfully log in in the browser.... And the fact that I am not running the browser from the same machine (browser on Win10 machine, your onedrive client on Debian machine)
Any other hint? Is there a URL checker somewhere that I can use with my URL to check what is wrong?

**Reply by abraunegg** [reply]
@darrepac 
> And the fact that I am not running the browser from the same machine (browser on Win10 machine, your onedrive client on Debian machine)

This will *not* be an issue - as this is how I also test. However, When using a browser etc like this, all the correct AzureAD tenant details is being handled by the browser.

> Any other hint? Is there a URL checker somewhere that I can use with my URL to check what is wrong?

For Personal Accounts, the response URI will be like this:
```
https://login.microsoftonline.com/common/oauth2/nativeclient?code=M.R3_BL2.RANDOM-UUID-VALUE
```
For normal Business Accounts / Office365 Accounts this will be:
```
https://login.microsoftonline.com/common/oauth2/nativeclient?code=RANDOM_AUTH_STRING_APPROX_793_CHARS&session_state=RANDOM-UUID-VALUE
```
For AzureAD tenants with custom configuration .... response could be anything.

You need to provide a full verbose debug log of your entire process via email. It could be that your University AzureAD has issues with the client, or it is expecting the client to have certain tenant details ...

You *may* need to configure the Linux client to use your University AzureAD tenant correctly. The steps are listed here: https://github.com/abraunegg/onedrive/blob/master/docs/national-cloud-deployments.md#step-5-configure-the-onedrive-client-to-use-the-specific-microsoft-azure-deployment

You are not using the 4 x Government Tenants, but you *may* need to configure the 'azure_tenant_id' configuration to allow the client to work with your University Account.

When authenticating the client use `--verbose --verbose --debug-https` to capture what is sent / received & what the application is doing when you are doing the authorise step + pasting in the browser response + what the application is doing when processing that response.

Archive & send this file to support@mynas.com.au

**Reply by darrepac** [reply]
ok so I have a "Business Accounts / Office365 Accounts"... I am going to email you the verbose output

**Comment by darrepac** [comment, marked as answer]
After direct support from Alex (thanks to him), trying to compile with DMD and LDC compiler without success, I tried the docker route and everything is working now
